conan/tools/gnu/autotools.py

```diff
@@ -75,28 +76,26 @@ def _fix_install_name(lib_name, lib_folder):
                                                                                          lib_name))
             self._conanfile.run(command)
 
-        def _is_modified_install_name(lib_name, lib_folder):
+        def _is_modified_install_name(lib_name, full_folder, libdir):
             """
             Check that the user did not change the default install_name using the install_name
             linker flag in that case we do not touch this field
             """
-            command = "otool -D {}".format(os.path.join(lib_folder, lib_name))
-            out = check_output_runner(command).strip().split(":")[1]
-            return False if str(os.path.join(lib_folder, shared_lib)) in out else True
+            command = "otool -D {}".format(os.path.join(full_folder, lib_name))
+            install_path = check_output_runner(command).strip().split(":")[1].strip()
+            default_path = str(os.path.join("/", libdir, shared_lib))
+            return False if default_path == install_path else True
 
         libdirs = getattr(self._conanfile.cpp.package, "libdirs")
-        for folder in libdirs:
-            full_folder = os.path.join(self._conanfile.package_folder, folder)
+        for libdir in libdirs:
+            full_folder = os.path.join(self._conanfile.package_folder, libdir)
             shared_libs = _osx_collect_dylibs(full_folder)
             for shared_lib in shared_libs:
-                if not _is_modified_install_name(shared_lib, full_folder):
+                if not _is_modified_install_name(shared_lib, full_folder, libdir):
                     _fix_install_name(shared_lib, full_folder)
 
     def install(self):
-        # FIXME: we have to run configure twice because the local flow won't work otherwise
-        #  because there's no package_folder until the package step
-        self.configure()
-        self.make(target="install")
+        self.make(target="install", args=["DESTDIR={}".format(self._conanfile.package_folder)])
```

Comment:
If one sets `default_configure_install_args` to `False`, they may not want the `DESTDIR` either.

Having used the tools in anger for a bit, having extra arguments added under the hood seems to be an anti pattern and hides the magic.

Personally, I would much prefer if `autotoolstoolchain` auto populated the `autoreconf_args`/`configure_args`/`make_args` and I could modify them as I see fit before generation.

Usage would then be:

```py
    def generate(self) -> None:
        at = AutotoolsToolchain(self)
        print(at.autoreconf_args) # has the `--force` and `--install` arguments in
        print(at.configure_args) # has the `--prefix` and `--shared` arguments in
        print(at.make_args) # has `DESTDIR`  and `-j` in it
        # I know _exactly_ will be passed to `autoreconf`, `configure` and `make`
```

I'd even argue adding `--enable-checking=fatal` would be a good default that users can easily remove with:

```py
        at.configure_args = [a for a in at.configure_args if a.startswith("--enable-checking")]
```

Then, if I have a really wonky package I can customise the arguments entirely. There are so many odd autotools packages in the wild.

Reply:
Thanks a lot for the review @mattyclarkson, I think something in that direction would make sense, I'll add some changes to see how it looks